transcom/mymove
# INT-B -19815- Office date validation/ alert - HHG, NTS, NTSR

## [B-19815](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A946352)

## Summary

**INTEGRATION**

This PR adds date validation for HHG, NTS and NTSR for requested pickup and/or delivery date selection for the shipment new/edit shipment views for SC and TOO roles.  It's a follow up to [PR-B -19813](https://github.com/transcom/mymove/pull/13426). An inline alert message will appear upon date selection from date picker. An alert message will appear on page load and on date selection. Message will be customized to respective date label. Alert will notify user if date selected is on weekend and/or holiday(federal holidays). For now implementation will assume date context is in the United States. For example, July 4th is a holiday and Sat/Sunday are considered weekend days.

Note: This PR contains code from [PR-B -19813](https://github.com/transcom/mymove/pull/13426)

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login as customer and create/submit HHG, NTS and NTSR shipment types for a given move. For each select appropriate pickup/delivery dates that will test weekend/holiday alerts.
2. Login as service counselor and verify dates that are weekend and/or holiday for requested pickup/delivery dates display appropriate warning alerts on view load in EDIT view.  Verify alert messages are specific to either pickup or delivery. It must match corresponding date input label.
3. Verify changing dates to non weekend/holiday dates do not display warning alerts...etc...go through all test cases.
4. Verify alerts work when newly HHG, NTS, NTSR are added in service counselor.
5. Update orders with all required fields and submit MOVE DETAILS.
6. Login as TOO and access submitted in step 5. Verify alerts work  for all shipment type created/updated for move in SC role. Alerts should behave the same as in SC.
7. Again verify alerts work when newly HHG, NTS, NTSR are added in TOO.  
8. Verify DateIsWeekendHoliday API in Office swagger.
9. Try to break it and go through all possible use cases you can think of!

### Frontend


### Backend


## Screenshots
